xc7/fasm2bels/iob_models.py

```diff
@@ -77,27 +77,52 @@ def append_obuf_iostandard_params(
         iostd_def = top.iostandard_defs[site.site.name]
 
         iostandard = iostd_def["IOSTANDARD"]
-        drive = iostd_def["DRIVE"]
+        drive = iostd_def["DRIVE"] if "DRIVE" in iostd_def else None
 
         # Check if this is possible according to decoded fasm
         is_valid = (iostandard, drive, slew) in possible_iostandards
         if not is_valid:
             print(
-                "IOSTANDARD+DRIVE+SLEW settings provided for {} do not match"
+                "IOSTANDARD+DRIVE+SLEW settings provided for {} do not match "
                 "their counterparts decoded from the fasm".format(
                     site.site.name
                 )
             )
+
+            print("Requested:")
+            print(" IOSTANDARD={}, DRIVE={}".format(iostandard, drive))
+
+            print("Candidates are:")
+            print(" IOSTANDARD        | DRIVE  | SLEW |")
+            print("-------------------|--------|------|")
+            for i, d, s in possible_iostandards:
+                print(
+                    " {}| {}| {}|".format(
+                        i.ljust(18),
+                        str(d).ljust(7), s.ljust(5)
+                    )
+                )
+
+            # Demote NSTD-1 to warning
```

Comment:
Why?

Reply:
The demotion happens only when the IOSTANDARD defined in JSON does not match to what can be decoded from FASM features. In that case the standard is not set at all.

I don't consider it a permanent solution but it makes Vivado tests pass (tests don't care about IOSTANDARDs untill they are not run in HW). A proper fix for that would require setting iostandards via eg. XDC file per design and then passing it to fasm2bels. That way we won't have mismatches.

Reply:
Given that we are working on XDC support and IOSTANDARD's, this issue should be resolved shortly.  Can you work with @tmichalak to make a detailed issue explaining what needs to be done?

Once the issue is created, please add a TODO here